File: drs/src/deprecated-test/com/db4o/test/replication/SingleTypeCollectionReplicationTest.java
/* Copyright (C) 2005   db4objects Inc.   http://www.db4o.com */

package com.db4o.test.replication;

import com.db4o.ObjectSet;
import com.db4o.drs.ReplicationSession;
import com.db4o.drs.inside.GenericReplicationSession;
import com.db4o.test.Test;

import java.util.Map;

public class SingleTypeCollectionReplicationTest extends ReplicationTestCase {

	protected void actualTest() {
		CollectionHolder h1 = new CollectionHolder();
		h1.map.put("1", "one");
		h1.set.add("two");
		h1.list.add("three");

		_providerA.storeNew(h1);
		_providerA.activate(h1);
		_providerA.commit();

		final ReplicationSession replication = new GenericReplicationSession(_providerA, _providerB);

		final ObjectSet objectSet = _providerA.objectsChangedSinceLastReplication();

		while (objectSet.hasNext()) {
			replication.replicate(objectSet.next());
		}

		replication.commit();

		ObjectSet it = _providerB.getStoredObjects(CollectionHolder.class);
		Test.ensure(it.hasNext());

		CollectionHolder replica = (CollectionHolder) it.next();
		Test.ensureEquals("one", replica.map.get("1"));
		Test.ensure(replica.set.contains("two"));
		Test.ensureEquals("three", replica.list.get(0));
	}

	public void test() {
		super.test();
	}
}

File: drs/drs.net/drs.tests.in/Db4objects/Db4o/Drs/Test/SingleTypeCollectionReplicationTest.cs
namespace Db4objects.Db4o.Drs.Test
{
    public class SingleTypeCollectionReplicationTest : Db4objects.Db4o.Drs.Test.DrsTestCase
    {
        protected virtual void ActualTest()
        {
            Db4objects.Db4o.Drs.Test.CollectionHolder h1 = new Db4objects.Db4o.Drs.Test.CollectionHolder();
            h1.ht["1"] = "One";
            h1.set["2"] = "two";
            h1.list.Add("three");
            A().Provider().StoreNew(h1);
            A().Provider().Commit();
            Db4objects.Db4o.DrsReplicationSession replication = new Db4objects.Db4o.Drs.Inside.GenericReplicationSession(A().Provider(), B().Provider());
            Db4objects.Db4o.ObjectSet objectSet = A().Provider().ObjectsChangedSinceLastReplication();
            while (objectSet.HasNext())
            {
                replication.Replicate(objectSet.Next());
            }
            replication.Commit();
            Db4objects.Db4o.ObjectSet it = B().Provider().GetStoredObjects(typeof(Db4objects.Db4o.Drs.Test.CollectionHolder));
            Db4oUnit.Assert.IsTrue(it.HasNext());
            Db4objects.Db4o.Drs.Test.CollectionHolder replica = (Db4objects.Db4o.Drs.Test.CollectionHolder)it.Next();
            Db4oUnit.Assert.AreEqual("One", replica.ht["1"]);
            Db4oUnit.Assert.IsTrue(replica.set.Contains("2"));
            Db4oUnit.Assert.AreEqual("three", replica.list[0]);
        }

        public virtual void Test()
        {
            ActualTest();
        }
    }
}
